tt_metal/common/core_coord.h

```diff
@@ -64,6 +64,93 @@ inline CoreCoord get_core_coord_from_relative(const RelativeCoreCoord &in, const
     return coord;
 }
 
+inline std::vector<CoreCoord> grid_to_cores(
+    uint32_t num_cores, uint32_t grid_size_x, uint32_t grid_size_y, bool row_wise = false) {
+    std::vector<CoreCoord> cores;
+    cores.reserve(num_cores);
+    TT_ASSERT(
+        num_cores <= grid_size_x * grid_size_y,
+        "Number of cores {} exceeds grid size {}x{}",
+        num_cores,
+        grid_size_x,
+        grid_size_y);
+    if (row_wise) {
+        for (uint32_t i = 0; i < num_cores; ++i) {
+            cores.push_back({i % grid_size_x, i / grid_size_x});
+        }
+    } else {
+        for (uint32_t i = 0; i < num_cores; ++i) {
+            cores.push_back({i / grid_size_y, i % grid_size_y});
+        }
+    }
+    return cores;
+}
+
+inline std::vector<CoreCoord> grid_to_cores(CoreCoord start, CoreCoord end, bool row_wise = false) {
```

Comment:
Why is implementation inline and in a header? 
Can you please remove inline from here and move implementation to cpp?

Reply:
I agree that they should be moved to cpp, but again I am just moving them around in the file just to get things functional.